src/test/java/com/rocketden/main/service/ProblemServiceTests.java

```diff
@@ -242,6 +248,28 @@ public void createTestCaseFailure() {
         assertEquals(ProblemError.EMPTY_FIELD, exception.getError());
     }
 
+    @Test
+    public void createTestCaseInvalidParsing() {
+        Problem expected = new Problem();
+        expected.setName(NAME);
+        expected.setDescription(DESCRIPTION);
+        expected.setDifficulty(ProblemDifficulty.HARD);
+        expected.addProblemInput(new ProblemInput("name", IO_TYPE));
+        expected.setOutputType(IO_TYPE_2);
+
+        CreateTestCaseRequest request = new CreateTestCaseRequest();
+        request.setInput(INPUT);
+        request.setOutput(OUTPUT);
+
+        Mockito.doReturn(expected).when(repository).findProblemByProblemId(expected.getProblemId());
+
+        ApiException exception = assertThrows(ApiException.class, () ->
+                problemService.createTestCase(expected.getProblemId(), request));
```

Comment:
**Note:** There are four other examples with this same fix below.

```suggestion
        String problemId = expected.getProblemId();
        ApiException exception = assertThrows(ApiException.class, () ->
                problemService.createTestCase(problemId, request));
```
This suggestion is based off the attached screenshot; since `expected.getProblemId()` could throw an error / exception as well, we should calculate that before so we can ensure that the error thrown comes from `createTestCase`.

<img width="768" alt="Exception of Inner Methods in Assert Throws" src="https://user-images.githubusercontent.com/7987279/106933066-58424900-66cd-11eb-96bb-9194dfa960f5.png">

Reply:
Sounds good - I made a separate commit for this since as you mentioned, there were other places that needed to be updated too